Sources/MapboxMaps/Annotations/ViewAnnotationOptions.swift

```diff
@@ -100,6 +100,36 @@ public struct ViewAnnotationOptions: Hashable {
         hasher.combine(offsetY)
         hasher.combine(selected)
     }
+
+    internal var frame: CGRect {
+        guard let width = width, let height = height else { return .zero }
+
+        let offset: (x: CGFloat, y: CGFloat) = (width * 0.5, height * 0.5)
+        var frame = CGRect(x: -offset.x, y: -offset.y, width: width, height: height)
+        let anchor = anchor ?? .center
+
+        switch anchor {
+        case .top:
+            frame = frame.offsetBy(dx: 0, dy: offset.y)
+        case .topLeft:
+            frame = frame.offsetBy(dx: offset.x, dy: offset.y)
+        case .topRight:
+            frame = frame.offsetBy(dx: -offset.x, dy: offset.y)
+        case .bottom:
+            frame = frame.offsetBy(dx: 0, dy: -offset.y)
+        case .bottomLeft:
+            frame = frame.offsetBy(dx: offset.x, dy: -offset.y)
+        case .bottomRight:
+            frame = frame.offsetBy(dx: -offset.x, dy: -offset.y)
+        case .left:
+            frame = frame.offsetBy(dx: offset.x, dy: 0)
+        case .right:
+            frame = frame.offsetBy(dx: -offset.x, dy: 0)
+        default: break
```

Comment:
Please, do not use the `default` case when possible. It would lead to potential issues when new options will be introduced. 
```suggestion
        case .center:
            fallthrough
        @unknown default:
            break
```